### PR TITLE
Device crd

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/proposals/device-crd.md [hunk 1]
```diff
@@ -89,74 +89,117 @@ A `device model` describes the device properties exposed by the device and prope
...
-  // Required.
-  Protocol     ProtocolType      `json:"protocol,omitempty"`
-  // A list of visitors describing the properties which can be accessed with the protocol.
-  // Required.
-  Visitors     []PropertyVisitor `json:"visitors,omitempty"`
+// Represents the type and data validation of a property.
+// Only one of its members may be specified.
+type PropertyType struct {
+	// +optional
+	Int    PropertyTypeInt64
```
Comment (rohitsardesai83) [inline docs/proposals/device-crd.md]: do we need json descriptors for all the properties ?

File: docs/proposals/device-crd.md [hunk 2]
```diff
@@ -232,27 +275,23 @@ A `device` instance represents an actual device object. It is like an instantiat
 ```go
 // DeviceSpec represents a single device instance. It is an instantation of a device model.
 type DeviceSpec struct {
```
Comment (rohitsardesai83) [inline docs/proposals/device-crd.md:283]: where can we store device instance specific details like device id , mac address etc ?

File: docs/proposals/device-crd.md [hunk 3]
```diff
@@ -280,25 +319,20 @@ type DeviceStatus struct {
...
 type Twin struct {
-  // The property name for which the expected/actual values are reported.
+  // Required: The property name for which the expected/actual values are reported.
   // This property should be present in the device model.
-  // Required.
   Name     string    `json:"name,omitempty"`
-  // the expected attribute value for this property.
-  // Required.
+  // Required: the expected attribute value for this property.
   Expected Attribute `json:"expected,omitempty"`
```
Comment (rohitsardesai83) [inline docs/proposals/device-crd.md]: should we rename Attribute to TwinProperty ?

File: docs/proposals/device-crd.md [hunk 4]
```diff
@@ -280,25 +319,20 @@ type DeviceStatus struct {
...
+  // Required: the actual attribute value for this property.
   Actual   Attribute `json:"actual,omitempty"`
 }
 
 // Attribute represents the device property for which an Expected/Actual state can be defined.
 type Attribute struct {
-  // Describes whether this property is optional or not.
-  // Required.
+  // Required: Describes whether this property is optional or not.
   Optional string            `json:"optional,omitempty"`
```
Comment (rohitsardesai83) [inline docs/proposals/device-crd.md]: I was thinking if we really need this field or not. Does it make sense ?

File: docs/proposals/device-crd.md (hunk 1 above)
Comment (kevin-wangzefeng) [reply docs/proposals/device-crd.md]: It's actually not required in a design doc, we can add them in the real code.

Comment (rohitsardesai83) [reply docs/proposals/device-crd.md]: ok got it

File: docs/proposals/device-crd.md (hunk 4 above)
Comment (kevin-wangzefeng) [reply docs/proposals/device-crd.md]: We don't need this field now

File: docs/proposals/device-crd.md (hunk 3 above)
Comment (kevin-wangzefeng) [reply docs/proposals/device-crd.md]: Yes we can

**Review by rohitsardesai83 — APPROVED** [review]
lgtm

**Pull request merged**